Dockerfile

```diff
@@ -2,19 +2,31 @@ FROM registry.access.redhat.com/ubi9/go-toolset:1.18.4-11 as builder
 
 USER 0
 WORKDIR /workspace
+
+ENV PATH="${PATH}:/opt/app-root/src/go/bin"
+RUN go env -w GO111MODULE=on
+
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+#RUN go mod tidy
 
 # Copy the go source
 COPY main.go main.go
 COPY internal/ internal/
 COPY config/ config/
 
+# Test
+#from builder as test
+
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build main.go
+#FROM builder as build
+COPY Makefile Makefile
+RUN go install github.com/onsi/ginkgo/v2/ginkgo@latest
+RUN go get github.com/onsi/gomega/...
+RUN go get -t github.com/RedHatInsights/xjoin-validation/internal/validator
```

Comment:
@fijshion I think would be better to put it inside Go dependencies?

For now, it is going to create more layers in the image, which is not ideal

Reply:
Ginkgo will need to be installed into the image's $PATH: https://onsi.github.io/ginkgo/#installing-ginkgo. I'm not sure what the $PATH variable looks like inside this image. I would check what's included in $PATH first, then either add the path to ginkgo to $PATH or install ginkgo inside the existing $PATH, whatever is easiest. I'm not sure if the binary is included via `go mod download` or if you need to manually install it.

I'm not sure what `go get -t github.com/RedHatInsights/xjoin-validation/internal/validator` is trying to do but it shouldn't be necessary. That should be included in the image via `COPY internal/ internal/`

Reply:
@fijshion Thanks!

For the ginkgo I think we are good! I'm adding $GOBIN to the images $PATH, and when installing it with the provided commands it works just fine (we just add 2 more layers to the image, and I think that it is okay!)

On the `/internal/validator/` this is what I receive when trying to trigger the test without the `go get -t github.com/RedHatInsights/xjoin-validation/internal/validator` command

```
content_test.go:4:2: missing go.sum entry for module providing package github.com/DATA-DOG/go-sqlmock (imported by github.com/RedHatInsights/xjoin-validation/internal/validator); to add:
        go get -t github.com/RedHatInsights/xjoin-validation/internal/validator
FAIL    github.com/RedHatInsights/xjoin-validation/internal/validator [setup failed]
FAIL
```

Reply:
this worked for me:

```
FROM registry.access.redhat.com/ubi9/go-toolset:1.18.4-11 as builder

USER 0
WORKDIR /workspace

ENV PATH="${PATH}:/opt/app-root/src/go/bin"
RUN go env -w GO111MODULE=on

# Copy the Go Modules manifests
COPY go.mod go.mod
COPY go.sum go.sum
# cache deps before building and copying source so that we don't need to re-download as much
# and so that source changes don't invalidate our downloaded layer
RUN go mod download

# Copy the go source
COPY main.go main.go
COPY internal/ internal/
COPY config/ config/

# Test
#from builder as test

# Build
#FROM builder as build
COPY Makefile Makefile
RUN go install github.com/onsi/ginkgo/v2/ginkgo@latest
#RUN go get github.com/onsi/gomega/...
#RUN go get -t github.com/RedHatInsights/xjoin-validation/internal/validator
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=mod main.go

#ENTRYPOINT ["/workspace/main"]
```

```
docker build . --tag validation:latest
docker run validation:latest ginkgo ./...
```

Reply:
Nice! Thanks, this looks much better!